db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/SPSPropertyRegistry.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2012 - 2020 Splice Machine, Inc.
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.db.impl.sql.execute;
+
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.reference.Property;
+import com.splicemachine.db.iapi.sql.compile.CompilerContext;
+import com.splicemachine.db.impl.services.uuid.BasicUUID;
+import com.splicemachine.db.impl.sql.compile.StatementNode;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A static registry that keeps track of all SPS properties and checks whether a dependency should be created
+ * from a statement plan to a specific SPS property.
+ *
+ * If you add a new SPS property, make sure to register it here so it is checked when analyzing a statement plan.
+ */
+public class SPSPropertyRegistry {
+    private static Set<SPSProperty> propertySet = ConcurrentHashMap.newKeySet();
+    static {
+        propertySet.add(new SPSPropertyTimestampFormat(new BasicUUID("91916f24-fa35-493d-9048-e41ffbe004d7"), Property.SPLICE_TIMESTAMP_FORMAT));
```

Comment:
Is this still a bit too aggressive? This should only affect Timestamp -> Char conversions, but this invalidates all statements with a timestamp reference if I understand it correctly. Would it be possible to restrict it further?

Reply:
I did some refactoring, please revise my PR message for more information about the new approach. Now, we scan exactly the `CAST` pattern from timestamp to string type.